src/ros/path_publisher.cpp

```diff
@@ -0,0 +1,100 @@
+#include <ros/ros.h>
+#include <nav_msgs/Path.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <tf/transform_listener.h>
+#include <vector>
+#include <string>
+#include "init_utils.h"
+
+static double tf_wait_for_transform_duration() {
+  double wait_for_transform_duration;
+  ros::param::param<double>("~ros/tf/wait_for_transform_duration", wait_for_transform_duration, 3.0);
```

Comment:
I suggest to remove prefixes from parameters names (e.g.  `~ros/tf/wait_for_transform_duration` -> `~wait_for_transform_duration`)